src/main/java/seedu/address/logic/commands/addcommands/AddTaskToGroupCommand.java

```diff
@@ -32,58 +35,64 @@ public class AddTaskToGroupCommand extends Command {
         + PREFIX_GROUP_NAME + "Team 1";
 
-    public static final String MESSAGE_SUCCESS = "Added task: %1$s to %2$s";
+    public static final String MESSAGE_SUCCESS = "Added task (%1$s) to the following groups:\n%2$s";
 
     public static final String MESSAGE_DUPLICATE_TASK_IN_GROUP = "This task is already in the group";
 
     private final TaskName taskName;
 
     private final Deadline deadline;
 
-    private final GroupName toAddInto;
+    private final Set<GroupName> toAddInto;
 
     /**
      * Creates an AddStudentToGroupCommand to add the specified {@code Task} to the specified {@code Group}.
      */
-    public AddTaskToGroupCommand(TaskName taskName, Deadline deadline, GroupName groupName) {
+    public AddTaskToGroupCommand(TaskName taskName, Deadline deadline, Set<GroupName> groupNames) {
         requireNonNull(taskName);
-        requireNonNull(groupName);
+        requireNonNull(groupNames);
         this.taskName = taskName;
         this.deadline = deadline;
-        this.toAddInto = groupName;
+        this.toAddInto = groupNames;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        if (!model.containsGroupName(toAddInto)) {
-            throw new CommandException(MESSAGE_GROUP_NAME_NOT_FOUND);
-        }
-
         Task task = new Task(taskName, deadline);
-        Group group = model.getGroupByName(toAddInto);
-
-        if (model.hasTaskInGroup(task, group)) {
-            throw new CommandException(MESSAGE_DUPLICATE_TASK_IN_GROUP);
-        }
-
-        model.addTaskToGroup(task, group);
-        if (!model.hasTask(task)) {
-            model.addTask(task);
-        } else {
-            model.increaseGroupWithTask(task);
+        List<GroupName> groups = toAddInto.stream().toList();
+        int lastIndex = groups.size() - 1;
+        StringBuilder groupsAdded = new StringBuilder();
+
+        for (GroupName groupName : groups) {
+            if (!model.containsGroupName(groupName)) {
+                throw new CommandException(MESSAGE_GROUP_NAME_NOT_FOUND);
+            }
+            Group group = model.getGroupByName(groupName);
+
+            if (model.hasTaskInGroup(task, group)) {
+                throw new CommandException(MESSAGE_DUPLICATE_TASK_IN_GROUP);
+            }
+            model.addTaskToGroup(task, group);
+            if (!model.hasTask(task)) {
+                model.addTask(task);
+            } else {
+                model.increaseGroupWithTask(task);
+            }
+            groupsAdded.append(groupName).append("\n");
         }
-        model.setMostRecentGroupTaskDisplay(group.getGroupName().fullName);
-        model.updateFilteredGroupList(x -> x.getGroupName().equals(group.getGroupName()));
+        model.setMostRecentGroupTaskDisplay(groups.get(lastIndex).getGroupName());
+        model.updateFilteredGroupList(x -> x.getGroupName().equals(groups.get(lastIndex)));
```

Comment:
Is it better to adjust the predicate so that either we display:

Task List Panel showing the task that has been added
or
Group List Panel showing all the groups that have this task being added in.

Reply:
ya, I was thinking if we can update display all tasks UI to show the groups, we should direct the user there after this command

but I followed the previous behaviour for this currently